### PR TITLE
Retry on "Too Many Requests" errors

### PR DESCRIPTION
**Pull request by tsujigiri**

### CODE REVIEW

File: spec/lib/easybill-rest-api/api_client_spec.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+                                            :headers => {},
+                                            :body => 'DATADATADATA',
+                                            :success? => true)
+    end
+
+    let(:request) { instance_double('Typhoeus::Request') }
+
+    it 'retries api calls that return a "Too Many Requests" error' do
+      allow(Retryable).to receive(:retryable).and_wrap_original do |m, opts, &block|
+        m.call(opts.merge(:sleep => 0), &block)
```
Comment (RKushnir) [inline spec/lib/easybill-rest-api/api_client_spec.rb]: You wouldn't need to do this if `sleep` wasn't hard-coded

**Comment by RKushnir** [comment]
:green_apple:

**Comment by dziemba** [comment]
How about using `alias_method_chain` (or the non-rails equivalent) so you don't have to change all the generated code? It also looks nicer than wrapping all that logic in a giant retry-block.

File: spec/lib/easybill-rest-api/api_client_spec.rb (hunk 1 above)
Comment (tsujigiri) [inline spec/lib/easybill-rest-api/api_client_spec.rb]: That's correct. So? :stuck_out_tongue: 

I'll change that...

**Comment by tsujigiri** [comment]
Good suggestion! Did that.

**Comment by tsujigiri** [comment]
Anything else?

File: lib/easybill-rest-client/configuration.rb [hunk 2]
```diff
@@ -166,5 +179,13 @@ def auth_settings
           },
       }
     end
+
+    def retry_cool_off_time
```
Comment (dziemba) [inline lib/easybill-rest-client/configuration.rb]: Why not stick to the scheme of setting default values in `#initialize`?

File: spec/lib/easybill-rest-api/api_client_spec.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
+RSpec.describe Easybill::ApiClient do
+  subject do
+    described_class.new(config)
+  end
+
+  let(:config) do
+    c = Easybill::Configuration.new
```
Comment (dziemba) [inline spec/lib/easybill-rest-api/api_client_spec.rb]: `Easybill::Configuration.new` already yields the config object, it looks a bit nicer than this.

File: spec/lib/easybill-rest-api/api_client_spec.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+                                            :success? => true)
+    end
+
+    it 'retries api calls that return a "Too Many Requests" error' do
+      allow(subject).to receive(:build_request).and_return(request)
+
+      expect(request).to receive(:run).and_return(too_many_requests)
+      expect(request).to receive(:run).and_return(ok)
+
+      expect(subject.call_api(:get, '/')).to eq([nil, 200, {}])
```
Comment (dziemba) [inline spec/lib/easybill-rest-api/api_client_spec.rb]: Shouldn't `DATADATADATA` be in here somewhere?

Comment (tsujigiri) [inline spec/lib/easybill-rest-api/api_client_spec.rb]: Not if I don't provide a return_type.

Comment (tsujigiri) [inline spec/lib/easybill-rest-api/api_client_spec.rb]: I provided a return_type. ;)

**Comment by tsujigiri** [comment]
Anything else?

**Comment by dziemba** [comment]
:mushroom:

**Pull request merged**